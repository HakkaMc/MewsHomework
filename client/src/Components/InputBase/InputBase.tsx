import React from "react";
import {
	Field,
	WrappedFieldProps,
	WrappedFieldInputProps,
	CommonFieldProps
} from "redux-form";
import { InputBase, InputBaseProps } from "@material-ui/core";

const Component: React.FC<WrappedFieldInputProps &
	InputBaseProps &
	CommonFieldProps &
	WrappedFieldProps> = (
	props: WrappedFieldInputProps &
		InputBaseProps &
		CommonFieldProps &
		WrappedFieldProps
) => {
	const updatedProps: any = {
		...props,
		...props.input
	};

	delete updatedProps.meta;

	return <InputBase {...updatedProps} />;
};

const Input: React.FC<WrappedFieldInputProps &
	InputBaseProps &
	CommonFieldProps &
	any> = props => <Field {...props} component={Component} />;

export default Input;
