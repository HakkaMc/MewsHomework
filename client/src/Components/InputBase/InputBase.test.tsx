import React from "react";
import { shallow } from "enzyme";

import InputBase from "./InputBase";

describe("<InputBase/>", () => {
	it("Match snapshot", () => {
		const wrapper = shallow(<InputBase />);
		expect(wrapper).toMatchSnapshot();
	});
});
