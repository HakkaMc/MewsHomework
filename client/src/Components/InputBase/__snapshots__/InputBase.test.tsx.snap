// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputBase/> Match snapshot 1`] = `
<Field
  component={[Function]}
/>
`;
